packages/camera/camera_avfoundation/example/ios/RunnerTests/CameraSettingsTests.m

```diff
@@ -0,0 +1,256 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@import camera_avfoundation;
+@import camera_avfoundation.Test;
+@import XCTest;
+@import AVFoundation;
+#import <OCMock/OCMock.h>
+#import "CameraTestUtils.h"
+#import "MockFLTThreadSafeFlutterResult.h"
+
+static const char *TEST_RESOLUTION_PRESET = "medium";
+static const int TEST_FPS = 15;
+static const int TEST_VIDEO_BITRATE = 200000;
+static const int TEST_AUDIO_BITRATE = 32000;
+static const bool TEST_ENABLE_AUDIO = YES;
+
+@interface CameraSettingsTests : XCTestCase
+@property(readonly, nonatomic) FLTCam *camera;
+@end
+
+@implementation CameraSettingsTests
+
+/// Expect that optional positive numbers can be parsed
+- (void)testParsingPositiveNumberOrNull {
+  // extern function from CameraPlugin.m
+  NSNumber *toPositiveOrNull(NSNumber * number);
+
+  NSNumber *negative = toPositiveOrNull(@(-100));
+  XCTAssertNil(negative, "negative number should be parsed as nil");
+
+  NSNumber *parsedNil = toPositiveOrNull(nil);
+  XCTAssertNil(parsedNil, "nil should be parsed as nil");
+
+  NSNumber *parsedFloat = toPositiveOrNull(@(3.7));
+  XCTAssertEqual(parsedFloat, @3, "float should be parsed as positive int");
+
+  NSNumber *parsedPositive = toPositiveOrNull(@(5));
+  XCTAssertEqual(parsedPositive, @5, "positive should be parsed as positive");
+
+  NSNumber *nullIsNull = toPositiveOrNull((NSNumber *)[NSNull null]);
+  XCTAssertNil(nullIsNull, "[NSNull null] should be parsed as nil");
+
+  NSNumber *nanIsNil = toPositiveOrNull(@(NAN));
+  XCTAssertNil(nanIsNil, "NAN should be parsed as nil");
```

Comment:
This is 6 different tests, so should be written as such. A single unit test should test one conceptual thing.

Reply:
Splitted into different tests